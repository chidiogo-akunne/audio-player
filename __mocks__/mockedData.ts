export const mockedData = {
  title: 'Nightlife',
  audio:
    'https://raw.githubusercontent.com/Learnfield-GmbH/CodingChallange/master/react%20native/simple%20audio%20player/data/Nightlife.mp3',
  cover:
    'https://raw.githubusercontent.com/Learnfield-GmbH/CodingChallange/master/react%20native/simple%20audio%20player/data/Nightlife.png',
  totalDurationMs: 15696
};
