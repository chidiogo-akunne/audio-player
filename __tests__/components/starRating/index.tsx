import StarRating from '@components/starRating';
import renderer from 'react-test-renderer';
import ThemeProvider from '@theme';
import { ContextThemeProvider } from '@context/index';
import { mockedData } from '__mocks__/mockedData';
import { fireEvent, render } from '@testing-library/react-native';

describe('Star rating', () => {
  const tree = renderer.create(
    <ThemeProvider>
      <ContextThemeProvider>
        <StarRating item={mockedData} />
      </ContextThemeProvider>
    </ThemeProvider>
  );
  it(`renders correctly`, async () => {
    await expect(tree).toMatchSnapshot();
  });
  it(`change rating`, async () => {
    const onPressMock = jest.fn();
    const { getByTestId } = render(
      <ThemeProvider>
        <ContextThemeProvider>
          <StarRating item={mockedData} />
        </ContextThemeProvider>
      </ThemeProvider>
    );
    await fireEvent.press(getByTestId('starRating'));
    expect(onPressMock.mock.calls.length);
  });
});
