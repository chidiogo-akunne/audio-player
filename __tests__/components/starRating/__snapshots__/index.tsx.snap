// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Star rating renders correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
          "flexDirection": "row",
        },
        undefined,
      ]
    }
    testID="starRating"
  >
    <View
      collapsable={false}
      pointerEvents="none"
      style={
        Object {
          "marginHorizontal": 5,
          "marginLeft": 0,
          "marginRight": 0,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={30}
        bbWidth={30}
        focusable={false}
        height={30}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 30,
              "width": 30,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={30}
      >
        <RNSVGGroup>
          <RNSVGPath
            d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
            fill={4278190080}
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      collapsable={false}
      pointerEvents="none"
      style={
        Object {
          "marginHorizontal": 5,
          "marginLeft": 0,
          "marginRight": 0,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={30}
        bbWidth={30}
        focusable={false}
        height={30}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 30,
              "width": 30,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={30}
      >
        <RNSVGGroup>
          <RNSVGPath
            d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
            fill={4278190080}
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      collapsable={false}
      pointerEvents="none"
      style={
        Object {
          "marginHorizontal": 5,
          "marginLeft": 0,
          "marginRight": 0,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={30}
        bbWidth={30}
        focusable={false}
        height={30}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 30,
              "width": 30,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={30}
      >
        <RNSVGGroup>
          <RNSVGPath
            d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
            fill={4278190080}
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      collapsable={false}
      pointerEvents="none"
      style={
        Object {
          "marginHorizontal": 5,
          "marginLeft": 0,
          "marginRight": 0,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={30}
        bbWidth={30}
        focusable={false}
        height={30}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 30,
              "width": 30,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={30}
      >
        <RNSVGGroup>
          <RNSVGPath
            d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
            fill={4278190080}
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      collapsable={false}
      pointerEvents="none"
      style={
        Object {
          "marginHorizontal": 5,
          "marginLeft": 0,
          "marginRight": 0,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={30}
        bbWidth={30}
        focusable={false}
        height={30}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 30,
              "width": 30,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={30}
      >
        <RNSVGGroup>
          <RNSVGPath
            d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
            fill={4278190080}
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;
